jscomp/core/js_exp_make.ml

```diff
@@ -999,6 +969,37 @@ let rec int_comp (cmp : Lambda.comparison) ?comment  (e0 : t) (e1 : t) =
   | _ ->          
     bool_of_boolean @@ bin ?comment (Lam_compile_util.jsop_of_comp cmp) e0 e1
 
+let bool_comp (cmp : Lambda.comparison) ?comment (e0 : t) (e1 : t) = 
+  match e0.expression_desc, e1.expression_desc with 
+  | Bool l, Bool r  ->
+    bool (match cmp with 
+    | Ceq  -> l = r
+    | Cneq -> l <> r
+    | Clt  -> l < r 
+    | Cgt  -> l > r 
+    | Cle  -> l <= r 
+    | Cge  -> l >= r
+    )
+  | Bool l, _  -> 
+    begin match cmp with 
+    | Clt -> seq e1 caml_false
```

Comment:
@bobzhang what's the idea here: why is this always returning false?
Same question for the similar cases below.